### ISSUE TITLE
[L4D1 – L4D2] Damaged and malfunctioning cvars

### ISSUE THREAD

**Issue by AshThe9thSurvivor** [ValveSoftware/Source-1-Games#3309]
We all know that the game can be modified to our liking, as much as if we are competitive players, or if we are administrators of community servers, thus giving a different and even better experience of the game, in the course of my years in the game I discovered that many cvars are damaged, in some cases they do not even work, perhaps nobody has asked them to fix this, but I will request it, it may be that here are not all the damaged cvars since they are many and I do not remember them all, but I will mention the ones that we use in my server and those that have caused me problems.

-	sv_downloadurl: I will start with the one that has caused problems for all L4D1 and L4D2 players, since there is no UI that shows the progress of the downloads from the server to the client, this exists in all games from the source engine, from Half-Life to CS: GO, for some reason it does not work in L4D, players only see a black screen and that scares them, they usually leave the servers, they do not even understand what is happening, and we only provide them with easy downloads of light maps, or characters new, even sounds, this has to be corrected, community server administrators suffer for it, keep in mind that we are the ones who keep the game active at the cost of our own money.

-	sv_glowenable: I don't know if this cvar exists in L4D2, but it works in singleplayer, even if you create games on local server, but it stops working on dedicated servers created by the community, this cvar helps us to remove the glow from survivors and objects , with this cvar we managed to make a realistic game in L4D, since you cannot see the comrades through the walls, I would appreciate repairing this cvar and I am sure that many also, since we are too many administrators who seek to remove the glow of the survivors and the items.

-	survivor_death_anims: This cvar adds a death animation to the survivors, it works on L4D1 and L4D2, the question is; and What is the error ?, because the animation works perfectly, but when a survivor about to die or in the middle of the death animation is attacked by a tank, this does not end his animation, so with each hit of the tank he flies aside and the animation restarts, if the tank corners the survivor in a corner, it can keep hitting the survivor for eternity, since the animation keeps restarting and never dies, this obviously breaks the game as the tank keeps hitting the immortal survivor while the others kill the tank, in some cases when the tank dies after hitting the survivor, this survivor no longer does the death animation and remains alive, fix this cvar, that animation adds realism to the game, and all players like it.

-	mapcyclefile: I think that many server administrators know this cvar, this makes the game not stop, every time a campaign ends, this cvar makes it continue to another, the one programmed by the administrator, although they do not believe it, this cvar does not work absolutely not at all, it is completely damaged, and it is strange since even within the game there are 2 files that are its links, the files in which you program the order of campaigns you want, the files are: mapcycle.txt and missioncycle.txt, also this cvar is closely related to these other cvars that are activated but do not work:
mp_maxrounds
mp_timelimit
mp_roundlimit
mp_roundtime
It means that it is not just a cvar that does not work, but rather that there are several that work together, of which some work, I think so, since when activating them the console shows it as activated, but since the main cvar does not work (mapcyclefile), these others become useless, since nothing is achieved, I would appreciate that you review all these cvars together, since as I said, they work together, if you activate the mapcyclefile, then you use all the aforementioned for their proper performance.

I have tried to explain everything as detailed and correct as possible, I hope you can understand it, I also hope they fix all these cvars, since they are very important and useful in the game, in addition to some breaking the gameplay due to its malfunction, if some know of more damaged cvars, please report it to this thread, it is time for you to do an update fixing things that will be able to be used in the game, I hope your prompt correction, thank you very much.

**Comment by kisak-valve** [comment]
Hello @AshThe9thSurvivor, in general, multi-issue issue reports are bad for any issue tracker. So, only the first issue will be tracked here.

**Comment by AshThe9thSurvivor** [comment]
> Hola @AshThe9thSurvivor,en general, los informes de varios problemas son malos para cualquier rastreador de problemas. Por lo tanto, sólo el primer número será rastreado aquí.

Thank you very much, so I have to create several Threads for each one of the errors?

**Comment by kisak-valve** [comment]
Yes, one issue per issue report is preferred.

**Comment by AshThe9thSurvivor** [comment]
> Yes, one issue per issue report is preferred.

Thank you very much, I will consider it for the next, and sorry for the mistake.

**Comment by lunatixxx** [comment]
I don't understand why a simple thing like that is still not implemented, the download ui.

**Comment by AshThe9thSurvivor** [comment]
> I don't understand why a simple thing like that is still not implemented, the download ui.

Exactly, I have a server in L4D1, and it is necessary to download additional content, besides it does not have an auto-download system like TF2, it does not have a download progress interface, in TF2 you only need to put the new things in the custom folder and automatically it is ready to download when the user enters the server.
